neat-screen.js

```diff
@@ -84,155 +83,217 @@ function NeatScreen (cabal) {
-        if (err) return
-        self.state.channels = channels
-        self.loadChannel('default')
-        self.bus.emit('render')
+    cabal.users.getAll(function (err, users) {
+      if (err) return
+      cabal.client.users = users
 
-        self.cabal.channels.events.on('add', function (channel) {
-          self.state.channels.push(channel)
-          self.state.channels.sort()
+      updateLocalKey()
+
+      cabal.users.events.on('update', function (key) {
+        // TODO: rate-limit
+        cabal.users.get(key, function (err, user) {
+          if (err) return
+          cabal.client.users[key] = Object.assign(cabal.client.users[key] || {}, user)
+          if (cabal.client.user && key === cabal.client.user.key) cabal.client.user = cabal.client.users[key]
+          if (!cabal.client.user) updateLocalKey()
           self.bus.emit('render')
         })
       })
 
-      self.cabal.users.getAll(function (err, users) {
-        if (err) return
-        state.users = users
-
-        updateLocalKey()
-
-        self.cabal.users.events.on('update', function (key) {
-          // TODO: rate-limit
-          self.cabal.users.get(key, function (err, user) {
-            if (err) return
-            state.users[key] = Object.assign(state.users[key] || {}, user)
-            if (self.state.user && key === self.state.user.key) self.state.user = state.users[key]
-            if (!self.state.user) updateLocalKey()
+      cabal.on('peer-added', function (key) {
+        var found = false
+        Object.keys(cabal.client.users).forEach(function (k) {
+          if (k === key) {
+            cabal.client.users[k].online = true
+            found = true
+          }
+        })
+        if (!found) {
+          cabal.client.users[key] = {
+            key: key,
+            online: true
+          }
+        }
+        self.bus.emit('render')
+      })
+      cabal.on('peer-dropped', function (key) {
+        Object.keys(cabal.client.users).forEach(function (k) {
+          if (k === key) {
+            cabal.client.users[k].online = false
             self.bus.emit('render')
-          })
+          }
         })
+      })
 
-        function updateLocalKey () {
-          self.cabal.getLocalKey(function (err, lkey) {
-            if (err) return self.bus.emit('render')
-            Object.keys(users).forEach(function (key) {
-              if (key === lkey) {
-                self.state.user = users[key]
-                self.state.user.local = true
-                self.state.user.online = true
-                self.state.user.key = key
-              }
-            })
-            self.bus.emit('render')
+      function updateLocalKey () {
+        cabal.getLocalKey(function (err, lkey) {
+          if (err) return self.bus.emit('render')
+          Object.keys(users).forEach(function (key) {
+            if (key === lkey) {
+              cabal.client.user = users[key]
+              cabal.client.user.local = true
```

Comment:
how is `cabal.client.user.local` used?

Reply:
I'm not sure if that is used. Do you know @noffle ?

Reply:
It's not used right now. For a bit I was experimenting with making the local user render differently on the nick sidebar.